libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -317,6 +316,18 @@ internal class MapboxTripSession(
                 isOffRoute = tripStatus.navigationStatus.routeState == RouteState.OFF_ROUTE
             }
         }
+
+        private fun calculateRemainingWaypoints(tripStatus: TripStatus): Int {
+            val routeCoordinates = tripStatus.route?.routeOptions()?.coordinatesList()
+            return if (routeCoordinates != null) {
+                val waypointsCount = routeCoordinates.size
+                val nextWaypointIndex = max(
+                    MIN_NEXT_WAYPOINT_INDEX,
+                    tripStatus.navigationStatus.nextWaypointIndex
+                )
```

Comment:
@VysotskiVadim would you be able to follow up with PR that adds a comment here explaining why we need a workaround and link to the fix that we're waiting for?

I think we should revert this workaround as soon as we get an attempted fix from Nav Native, otherwise, we could be obfuscating even more issues for multi-leg routes that will reroute incorrectly.

Reply:
@LukasPaczos https://github.com/mapbox/mapbox-navigation-android/pull/5276